x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/process/autodetect/AutodetectBuilder.java

```diff
@@ -83,8 +83,15 @@
     /**
      * The maximum number of anomaly records that will be written each bucket
      */
+    @Deprecated
     public static final Setting<Integer> MAX_ANOMALY_RECORDS_SETTING = Setting.intSetting("max.anomaly.records", DEFAULT_MAX_NUM_RECORDS,
-            Setting.Property.NodeScope);
+            Setting.Property.NodeScope, Setting.Property.Deprecated);
+    public static final Setting<Integer> MAX_ANOMALY_RECORDS_SETTING_DYNAMIC = Setting.intSetting(
```

Comment:
It would be good to add a comment that although this is now dynamic, updates only apply to newly opened jobs, not already running jobs.

I don't think this setting will be changed often enough that this is worth fixing.  (At the moment the setting is completely undocumented, so it's unlikely anyone will be changing it.)  But it's good to have a comment pointing out the limitation so that if we ever do decide to document it then the limitation will hopefully get documented too.